framework/contrib/hit/parse.cc

```diff
@@ -1848,10 +1848,10 @@ buildHITTree(std::shared_ptr<wasp::DefaultHITInterpreter> interpreter,
   if (hnv_parent.is_null())
     return;
 
-  for (std::size_t i = 0, count = hnv_parent.child_count(); i < count; i++)
+  for (auto itr = hnv_parent.begin(); itr != hnv_parent.end(); itr.next())
```

Comment:
Any reason the `wasp::HITNodeView` Iterator increments with `.next()` rather than with `++`? Because the latter would allow the use of range-based for loops.

Reply:
@dschwen -

I can overload the `++` operator in WASP to call `next()` if needed.

Reply:
@dschwen -

I overloaded the `++` and `*` operators on this iterator in WASP that are needed for range-based for loops.

After that is merged into the public repo, I'll update the WASP submodule here and change that for loop to

```
for (const auto & hnv_child : hnv_parent)
```

Reply:
@dschwen -

I just updated the WASP submodule and that for loop to be ranged-based in this commit - https://github.com/idaholab/moose/pull/24435/commits/3aefa45954f4c221f502cbe1ac4c06cbbfcae86a.